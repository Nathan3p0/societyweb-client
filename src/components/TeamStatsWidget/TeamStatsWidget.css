.teamstats__widget {
    background: #fff;
    margin: 1rem 0;
    box-shadow: 1px 3px 10px #33333363;
}

.teamstats__widget--header {
    background-color: #26495E;
    color: white;
    padding: .3rem;
    padding-left: 1rem;
    font-size: 1.5rem;
}

.teamstats__widget--content {
    padding: 1rem;
    font-size: 1.1rem;
    color: rgb(34, 34, 34);
}

.teamstats__widget--content h3 {
    font-size: 1.5rem;
}

.teamstats__widget--content h4 {
    font-size: 1.2rem;
}

.teamstats__widget--count {
    font-size: 1.4rem;
    margin-left: .4rem;
}