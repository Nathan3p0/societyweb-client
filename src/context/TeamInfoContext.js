import React from 'react';

const TeamInfoContext = React.createContext();

export default TeamInfoContext;