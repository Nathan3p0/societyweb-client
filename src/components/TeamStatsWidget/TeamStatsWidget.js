import React, { useContext } from 'react';
import TeamInfoContext from '../../context/TeamInfoContext';
import './TeamStatsWidget.css';

const TeamStatsWidget = (props) => {
    const team = useContext(TeamInfoContext);

    return (
        <section className="teamstats__widget">
            <h3 className="teamstats__widget--header">{team.teamName}</h3>
            <div className="teamstats__widget--content">
                <h3>Welcome {team.memberName}!</h3>
                <h4>Current Event Count: <span className="teamstats__widget--count">{team.events.length || 0}</span></h4>
                <h4>Active Member Count: <span className="teamstats__widget--count">{team.members.length || 0}</span></h4>
            </div>
        </section>
    );
}

TeamStatsWidget.defaultProps = {
    admin: 'Loading'
}

export default TeamStatsWidget;