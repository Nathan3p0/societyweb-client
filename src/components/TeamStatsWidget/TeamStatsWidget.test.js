import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import TeamStatsWidget from './TeamStatsWidget'
import TeamInfoContext from '../../context/TeamInfoContext';

describe('Team Stats Widget Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <TeamInfoContext.Provider value={{
                teamName: '',
                events: [],
                members: []
            }}>
                <TeamStatsWidget />
            </TeamInfoContext.Provider>
            , div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(
            <TeamInfoContext.Provider value={{
                teamName: '',
                events: [],
                members: []
            }}>
                <TeamStatsWidget />
            </TeamInfoContext.Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot()
    })
})