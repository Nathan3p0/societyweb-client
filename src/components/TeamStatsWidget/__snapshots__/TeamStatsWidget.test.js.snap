// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team Stats Widget Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="teamstats__widget"
>
  <h3
    className="teamstats__widget--header"
  >
    
  </h3>
  <div
    className="teamstats__widget--content"
  >
    <h3>
      Welcome 
      !
    </h3>
    <h4>
      Current Event Count: 
      <span
        className="teamstats__widget--count"
      >
        0
      </span>
    </h4>
    <h4>
      Active Member Count: 
      <span
        className="teamstats__widget--count"
      >
        0
      </span>
    </h4>
  </div>
</section>
`;
